modules/geo-layers/src/mesh-layer/mesh-layer.js

```diff
@@ -28,6 +31,23 @@ export default class _MeshLayer extends SimpleMeshLayer {
     return {...shaders, vs, fs};
   }
 
+  initializeState() {
+    const {attributeManager} = this.state;
+    const {pickFeatures, objectIds} = this.props;
+    super.initializeState();
+
+    if (pickFeatures && objectIds) {
```

Comment:
If you allow the user of this layer to switch picking mode then this attribute should always be added

Reply:
Sometimes i3s geometry doesn't have `objectIds` at all. It that case we should check it.

Reply:
My comment is regarding `pickFeatures`

Reply:
Not really clear understand this comment but let me explain possible picking cases:
All cases works only if `picking` flag is true!

1. `ObjectIds` exists and `pickFeatures` is `on` - feature picking works
2. `ObjectIds` exists and `pickFeatures` is `off` - instance picking works
3. `ObjectIds` don't exist and `pickFeatures` is `on` - instance picking works
4. `ObjectIds` don't exist and `pickFeatures` is `off` - instance picking works

I see that you want to avoid multiple checking of `pickFeatures` and `objectIds` but I can't manipulate only one flag here.
I don't want to create `ObjectIds` always on the loaders.gl side if there are no feature ids in the geometry.

Reply:
Yes, you were right. I decided to keep only `featureIds` array. I'll organize switching between `instancePickingColors` and `featurePickingColors` on loaders.gl side, just for provide `featureIds` or not.